beacon-chain/blockchain/process_block_helpers.go

```diff
@@ -356,3 +356,32 @@ func (s *Service) ancestor(ctx context.Context, root []byte, slot uint64) ([]byt
 
 	return s.ancestor(ctx, b.ParentRoot, slot)
 }
+
+// This updates justified check point in store, if the new justified is later than stored justified or
+// the store's justified is not in chain with finalized check point.
+//
+// Spec definition:
+//   if (
+//            state.current_justified_checkpoint.epoch > store.justified_checkpoint.epoch
+//            or get_ancestor(store, store.justified_checkpoint.root, finalized_slot) != store.finalized_checkpoint.root
+//        ):
+//            store.justified_checkpoint = state.current_justified_checkpoint
+func (s *Service) finalizedNewJustified(ctx context.Context, state *stateTrie.BeaconState) error {
```

Comment:
can we rename to something more informative? finalizedImpliesNewJustifiedCheckpoint or something like that